lib/decompress/zstd_decompress.c

```diff
@@ -1633,10 +1652,15 @@ size_t ZSTD_decompressStream(ZSTD_DStream* zds, ZSTD_outBuffer* output, ZSTD_inB
             RETURN_ERROR_IF(zds->fParams.windowSize > zds->maxWindowSize,
                             frameParameter_windowTooLarge);
 
+            ZSTD_updateOversizedDuration(zds);
             /* Adapt buffer sizes to frame header instructions */
             {   size_t const neededInBuffSize = MAX(zds->fParams.blockSizeMax, 4 /* frame checksum */);
                 size_t const neededOutBuffSize = ZSTD_decodingBufferSize_min(zds->fParams.windowSize, zds->fParams.frameContentSize);
-                if ((zds->inBuffSize < neededInBuffSize) || (zds->outBuffSize < neededOutBuffSize)) {
+
```

Comment:
Maybe easier if you just call `ZSTD_updateOversizedDuration()` duration here and pass `neededInBuffSize` and `neededOutBuffSize` in as arguments. That way we don't calculate them again in the function body in a way that could possibly become incorrect if someone updated this calculation here but not the one in the function.